### PR TITLE
Refine rename flow for blocks with overrides

### LINKED ISSUE #60118: Remove duplicative "Rename" control from the Block Inspector
The "Advanced" panels in the Block Inspector have become a catch-all of sorts; I'd like to clean up and organize what we have. 

The first thing that comes to mind is the duplicative "Block Name" control. Renaming currently happens for blocks within the block toolbar, or in List View. 

It's much easier, and requires less steps, from either—and both open the same familiar modal for renaming blocks (which is also used for templates, template parts, patterns, and perhaps pages soon). 

This control populated in each block's "Advanced" panel is the only of its kind for renaming. I think we should remove it and keep the rename flow consistent across the editing experience. 

### Current 

CleanShot 2024-03-22 at 10 52 29

### PR DESCRIPTION
**Pull request by kevin940726**
## What?

Part of https://github.com/WordPress/gutenberg/issues/59813. Supersedes https://github.com/WordPress/gutenberg/pull/60066. Also close https://github.com/WordPress/gutenberg/issues/60118.

**The latest design is https://github.com/WordPress/gutenberg/pull/60234#issuecomment-2034513896**

This PR does a few things:
1. If no name exists on block, and overrides is disabled, use "allow overrides" button
2. If name exists and overrides is disabled, use toggle
3. If overrides is enabled, use toggle
4. When clicking button, show "allow overrides" modal that is essentially a rebranded "rename block" modal
5. In all other rename actions, use the standard rename modal with overrides warning text
6. The default state for overrides is disabled, users have to opt-in to enable pattern overrides

## Why?

This is to simplify the UI flow for renaming and toggling overrides for blocks.

## How?

Add overrides warning to the rename modal and change the overrides control to a button when there's no name.

## Testing Instructions

1. Create a synced pattern and add a few paragraphs.
2. If the block has no name, the control should show a button for overrides.
3. Clicking on the button should open a renaming modal.
4. Once the block is named, the control becomes a toggle button.
5. Other renaming flow (list view, block toolbar, shortcut) should show overrides warning.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/7753001/37ed9f09-0b7b-447d-835f-00dcabc98077

### CODE REVIEW

**Comment by jasmussen** [comment]
Nice work, although there are some tweaks to be done here and there, this seems to have the right ingredients. Can you [have a look at this one](https://github.com/WordPress/gutenberg/issues/59813#issuecomment-2029417050) and see if that helps thread the needle? There may be some simplified name management by unifying the modal.

File: packages/block-editor/src/components/block-rename/modal.js
Comment (talldan) [inline packages/block-editor/src/components/block-rename/modal.js]: I personally don't think this checkbox is needed in the experience. Obvious explanatory text feels like enough to me. Another additional step could be to make the button use the 'destructive' style.

Might need some second opinions from designers.

File: packages/block-editor/src/components/block-rename/modal.js
Comment (talldan) [inline packages/block-editor/src/components/block-rename/modal.js:27]: I don't think the generic `block-editor` package shouldn't contain code that reasons about pattern overrides, which is a WordPress feature.

It's tricky because we're leveraging block naming, which is a generic block editor feature. 

I wondered if we could use an editor setting to disable the regular block naming UI and replace it with a custom one (possibly implemented in the `editor` package) when a user is editing a pattern.

File: packages/block-editor/src/components/block-rename/modal.js
Comment (talldan) [inline packages/block-editor/src/components/block-rename/modal.js]: A user probably wouldn't know what we mean about 'connections', so I'm wondering if there's a better way of phrasing this.

Linking to a help article could also be an option in the long term. Not something we have right now, but once the flow is a little more concrete it might be a good idea to add one explaining pattern overrides

File: packages/patterns/src/components/allow-overrides-modal.js
Comment (talldan) [inline packages/patterns/src/components/allow-overrides-modal.js]: I think this text is a little informal for WordPress copy. Probably ok to stick with 'Block name'.

File: packages/block-editor/src/hooks/block-renaming.js
Comment (talldan) [inline packages/block-editor/src/hooks/block-renaming.js]: Again this copy feels like it could be improved.

**Review by talldan — COMMENTED** [review]
This works pretty well.  I left a few PR comments, mainly about the copy which could do with some adjustments.

In testing, there's one flow that doesn't seem to be covered:
1. Create a pattern
2. Add a paragraph
3. Use the block options dropdown menu (either via List View or the canvas) to name a block.

You see a different UI in the modal when doing this, but still have overrides enabled after naming the block. Feels like there might be two options to solve this:
- Show 'Allow overrides' in the dropdown instead of 'Rename', and use the same modal in all places (when editing a pattern). That way everywhere has the same flow.
- (Or) allow the user to name block as normal, but don't enable overrides. Instead, still require the user to click 'Allow overrides' in the advanced area.

In practice, the 'Allow overrides' button that turns into a toggle also doesn't feel like the best option to me, but it seems more of a minor thing that can alway be iterated on.

**Comment by jasmussen** [comment]
Took this one for a spin, and I'm missing a little nuance, seeing the toggle but no block name most of the time, and I wasn't able to find the naming button. It's very probably I'm doing something wrong. I would appreciate a quick review by @SaxonF

**Comment by kevin940726** [comment]
> You see a different UI in the modal when doing this, but still have overrides enabled after naming the block. Feels like there might be two options to solve this

Yeah, this is based on the assumption that naming a block automatically enables overrides. You need to "opt-out" of the feature manually after naming it (https://github.com/WordPress/gutenberg/issues/59812). (I have multiple explorative PRs all connecting together, and I know this is kind of confusing 😅)

> seeing the toggle but no block name most of the time, and I wasn't able to find the naming button. 

The naming control in the advanced panel is removed by design. This also matches the proposal in https://github.com/WordPress/gutenberg/issues/60118. This is just one possible option though and we can revisit this separately. If you can't find the rename button in the dropdown then it's likely a bug though 😅.

**Comment by SaxonF** [comment]
> Yeah, this is based on the assumption that naming a block automatically enables overrides. You nee

I would not automatically enable syncing via renaming, at least initially. The direction this PR goes in is adding an explicit way to enable overrides (vs based on name) whilst encouraging a name change, which I think is the right approach at least in the short term. We keep the renaming action and overrides action separate, but can tie them together via the "allow overrides" button.

What do we need to do to get this over the line? I think its a big improvement over what we have and is a sensible start point. Keen to get @fabiankaegy thoughts on this too.

**Comment by talldan** [comment]
Just to flag some dependencies. I think this design direction (opt-in to block naming) has a knock on effect with this PR - [Use metadata.name only as the hint for pattern overrides](https://github.com/WordPress/gutenberg/pull/59956#top). I think it makes it unworkable as a solution, so it'd mean going back to the drawing board on a solution for the issue that it would have solved - [Pattern Overrides: individual binding attributes are saved in the pattern markup and don't auto-update](https://github.com/WordPress/gutenberg/issues/58601#top). Does that conclusion sound right, @kevin940726?

For me, it's not a blocker, but I want to be clear about what the interconnected tasks are.

**Comment by fabiankaegy** [comment]
Thanks for the ping :) This tests well! Kudos to everyone involved. 

I still stand by my strong opinion that the overrides toggle should not be enabled by default when a block gets renamed and instead decouple that action so that a user has to specifically opt-in. 

Besides that I noticed a few rough edges:
 - When a block is already named before the pattern gets created, it isn't editable in the instance till I reload the page. Seems the instance overrides are not initialized on pattern creation.
 - When a block already has a name & overrides enabled the block name setting is completely removed from the Advanced Settings and only reachable through the block context menu. I think this is confusing. I think the flow @jasmussen created here https://github.com/WordPress/gutenberg/issues/59813#issuecomment-2029417050 would work better for this
 - I'm also getting an error on this branch whenever I try to select a group block that is for example coming from tt4 which uses the pattern shuffling experiment. Not sure whether that is related.

**Comment by SaxonF** [comment]
> When a block already has a name & overrides enabled the block name setting is completely removed from the Advanced Settings and only reachable through the block context menu. I think this is confusing.

@fabiankaegy The block name was removed completely from the advanced panel so not just when override is enabled. See https://github.com/WordPress/gutenberg/issues/60118#top . There are a couple reasons for this that the issue touches on but the one related to this work is that renaming has the side effect of resetting instances that already override the block. This is a destructive action which is hard to provide guardrails around if its simply an input within the inspector with no confirm action.

The other main purpose of this exploration is clearly separating "rename" and "allow overrides" actions, hence not including toggle within the rename modal. Rename modal can be consistent whether its used in the source or on a random block in the editor. Relabelling the initial "allow overrides" button in the inspector to "set up overrides..." which triggers the "set up overrides" modal may help clarify why it turns to a toggle.

**Comment by fabiankaegy** [comment]
@SaxonF thanks for those insights. I wonder whether completely disabling the rename field may be a better solution. Removing it makes it seem like there is a bug

**Comment by kevin940726** [comment]
I updated the branch to add an "Allow overrides" toggle in the renaming modal when editing inside a pattern. I also made the toggle "opt-in" instead of opt-out to avoid some of the issues mentioned above.

https://github.com/WordPress/gutenberg/assets/7753001/017a18ef-3998-4017-ab33-32df2b614827

> [!NOTE]
> The code for pattern overrides probably shouldn't belong in the `block-editor` package. But for the sake of this exploration, I decided to refactor that in a future follow-up once we can confirm the design UX.

**Comment by SaxonF** [comment]
@fabiankaegy the name field shouldn't be there at all , are you seeing it visible on this branch in some cases? Consider removing block name from advanced a completely separate issue that also benefits this work .

**Comment by SaxonF** [comment]
@kevin940726 as noted above, I'm not convinced we need the overrides toggle in the rename modal and would prefer to keep rename and overrides separate actions for now. If we get feedback that people are looking for overrides in rename or related we can reassess.

**Comment by fabiankaegy** [comment]
> @fabiankaegy the name field shouldn't be there at all , are you seeing it visible on this branch in some cases? Consider removing block name from advanced a completely separate issue that also benefits this work .

The name field is visible in the advanced section of any block as of WordPress 6.5. I think that this ship has sailed. I would be very careful about removing it again from there because users are now getting used to it and I think it would be quite disruptive to remove it again.

**Comment by jasmussen** [comment]
Just chatted with Saxon about the feedback here, to better understand some of the path being explored in this PR. The short summary revolves around these two points:

1. Removing the block name field from the Advanced section.
2. The approach to "setting up block overrides".

I'll dive in with more details from that chat, but do it over a quick visual of the current state of this branch, with some points intermixed as feedback. GIF:
branch

The GIF shows editing a synced pattern, selecting an image, then clicking the "Allow overrides" button.

This button is meant to be a "setup" feature for setting up overrides. The motivation here is that allowing parts of the pattern is a bit of an advanced flow, so the modal can provide contet.

But as soon as a user-supplied name is provided, then the "allow overrides" just becomes a toggle.

Separately I try to rename a group block, just in the list view — turns out there's a bug there, the name for the group block is not reflected in the inspector.

**Naming**

Unpacking this — the block needs a name when overridden. Moving this into the modal invoked by the "Allow overrides" button makes it possible to provide context for why it needs a name.

There's also just the ergonimics of renaming a group block, _without_ enabling any overrides. This is a purely cosmetic change that is visible primarily in the list view. Here's an example of [the 6.5 release site](https://wordpress.org/download/releases/6-5/) structure:

6-5-release-site-list-view

Every module in that page is a Group block, and with a lot of content to build and edit, those cosmetics provide a meaingful overview and navigational affordance for anyone editing. But only really, in the list view. And since that's also the primary interface for renaming (ellipsis menu > rename), the fact that you can rename also in the Advanced section, is a little duplicative.

To that end, I can get behind the idea of removing the block name from the advanced section, and having that ellipsis menu be the primary place where you do that. 

So I can be on board with this PR as a step forward: having the button to initially provide a name for overrides makes sense. Removing the block name from advanced, also makes sense. There may be room to revisit the toggle that appears afterwards could be something to explore, but as-is it does provide a simple way to at a glance tell whether something is overridable or not, so it seems worth trying.

**Next steps**

Based on the above, I'd suggest the following. Firstly, to fix the bug where you can't rename a group block — not sure why that broke.

Separately, let's massage these modals:

allow overrides
rename
rename with override warning

Those last two are the same modal, the latter just has a contextual warning for when the the block allows overrides. Let's remove the checkbox and make them look more alike. A suggestion for the above three:

Modal tweaks

**Comment by fabiankaegy** [comment]
Thanks @jasmussen and @SaxonF :) That overview is very helpful. 

I stand by my comment from a moment ago in that I think it would be a wiser choice to disable the name field in the advanced section when overrides are active and maybe trigger the modal when a user attempts to click on it instead of removing it altogether.

Besides that one detail, I really think this flow that you are outlining is going to work very well :)

**Comment by SaxonF** [comment]
@fabiankaegy i don't see removing name input as a blocker for this work so could just provide warning text under input in short term and discuss name field in Rich's separate issue.

**Comment by fabiankaegy** [comment]
Also added a comment to the ticket you mentioned https://github.com/WordPress/gutenberg/issues/60118#issuecomment-2034582793

I think taking it out of this PR and then seeing whether we can come up with an alternative before we remove it altogether would be better. But I won't block progress here :) We have some time before the next WP core release to get this right

**Comment by richtabor** [comment]
> @fabiankaegy i don't see removing name input as a blocker for this work so could just provide warning text under input in short term and discuss name field in Rich's separate issue. 

Agreed. 

> maybe trigger the modal when a user attempts to click on it instead of removing it altogether.

We shouldn't trigger a model on input selection. And I also don't think having an input that is sometimes there, isn't ideal. 

I do think the Advanced panel is more confusing having the name input there, where it's largely irrelevant (you rename for List View). But let's carry on that conversation in https://github.com/WordPress/gutenberg/issues/60118

**Comment by kevin940726** [comment]
Updated again to match the design in https://github.com/WordPress/gutenberg/pull/60234#issuecomment-2034513896. Removed the toggle button and added back the warning text.

https://github.com/WordPress/gutenberg/assets/7753001/37ed9f09-0b7b-447d-835f-00dcabc98077

I guess the main concern is whether the warning text is enough for such a destructive action. I also change the save button to be destructive. We can tweak the design there though.

Please still consider this PR a WIP. I haven't added back the naming field in the advanced panel yet. But feel free to play with the main flow to test it out!

File: packages/block-editor/src/hooks/block-renaming.js
Comment (talldan) [inline packages/block-editor/src/hooks/block-renaming.js]: I believe the bug @jasmussen spotted with the group block is because this code needs to remain.

I've made a separate PR now that removes just the control, so I think we should be able to consider it a completely separate task:
[Remove block renaming control from advanced inspector controls group](https://github.com/WordPress/gutenberg/pull/60453#top)

**Comment by talldan** [comment]
I like the way this is heading. It feels fairly straightforward and shouldn't have any backwards compatibility concerns. As per previous feedback, I'm still unsure on the button that morphs into a toggle, but it's a minor thing that can be iterated on.

**Comment by jasmussen** [comment]
Nice progress. Two things unrelated to this PR:

1. In this video I notice the "The backup of this post is different". I've noticed this yellow notice popping up a great deal of times working with synced patterns lately. Since you're doing nothing wrong in that particular flow, something must be off with the page backup that we should look into.
2. I've noted in a few other unrelated threads that the isDestructive property for inputs and buttons is of dubious value. It's not useful to people with color-blindness, red is also culturally not always a useful color to denote "destructiveness", and the concept itself of being destructive is often nuanced when there are page revisions, etc. In this particular case, the modal _itself_ is the warning, and there's expansive text right above. That's just to say I'd love to never use isDestructive in modal _confirm_ dialogs, and in general use the prop sparingly. Could be useful to update the docs.

**Comment by kevin940726** [comment]
I rebased the branch to include https://github.com/WordPress/gutenberg/pull/60453 and removed the `isDestructive` prop.

**Comment by jasmussen** [comment]
Looks good. I still can't rename the Group block in this branch.

Edit: It works after all, I had incorrectly tested the rebased branch. Works well!

File: packages/block-editor/src/components/block-rename/modal.js
Comment (kevin940726) [reply packages/block-editor/src/components/block-rename/modal.js:27]: The more I think about it, the more I think this might be the simplest option 😅. The block-editor package also contains code for the block binding feature anyway. I think we can add a comment above it but maybe we don't need to extract it out, at least for now. WDYT?

Comment (talldan) [reply packages/block-editor/src/components/block-rename/modal.js:27]: I haven't thought of any alternative, either.

> The block-editor package also contains code for the block binding feature anyway.

I guess block bindings could also be considered a general block editor feature, though an implementer would have to bring part of their own implementation (all the dynamic rewriting of blocks on the frontend). 🤔 

My view is that Custom Fields, Pattern Overrides are more the WordPress specific things built on top of it.

> I think we can add a comment above it but maybe we don't need to extract it out, at least for now. WDYT?

Making an issue for it, adding it to the main Tracking/Epic/Iteration for pattern overrides would be good as well as the comment. I think a comment is too easily forgotten. 😄

Comment (kevin940726) [reply packages/block-editor/src/components/block-rename/modal.js:27]: Good point. Made an issue here: https://github.com/WordPress/gutenberg/issues/60619.

**Comment by fabiankaegy** [comment]
I just took this for a spin again and I really like how this works now.


https://github.com/WordPress/gutenberg/assets/20684594/c4836174-645c-4ccd-8f9c-acb737d7a5fe

Some small notes that I don't think are blocking :) Great work on getting this to this state 🚀 I'm really excited about this feature!

 1. The warning in the notice of the rename flow after a block has been marked to allow overrides is not noticeable/clear enough. I think that given the consequences of renaming a synced pattern the UX we had before with the checkbox is providing a better experience because it prevents very disruptive changes from happening unintentionally.  
image

2. What's the reason we are showing the block connection icon when an overwritable block is selected in the `contentOnly` locked mode? Seems odd to have this block icon shown here especially since it isn't very descriptive.
CleanShot 2024-04-10 at 07 54 24@2x

File: packages/patterns/src/components/allow-overrides-modal.js
Comment (talldan) [inline packages/patterns/src/components/allow-overrides-modal.js]: Some small suggestions here:
- We probably want to encourage users to properly capitalize the names
- The way it states 'Recipe' twice, I don't think users would be creating a 'Recipe' block in a 'Recipe' pattern, more likely a heading for the title or similar.
```suggestion
							'This name will be used to denote the override wherever the synced pattern is used. The name here will help people understand its purpose. E.g. if you\'re creating a recipe pattern, it can be "Recipe Title", "Ingredients", etc.'
```

File: packages/block-editor/src/components/block-rename/modal.js
Comment (talldan) [inline packages/block-editor/src/components/block-rename/modal.js]: I still think we should be letting users know what exactly will break in this message, and I don't think most users will know what the 'connection' part refers to:
```suggestion
										'This block allows overrides. Changing the name can cause problems with content entered into instances of this pattern.'
```

**Comment by jasmussen** [comment]
I'd avoid the checkbox for a few reasons:

* It's an advanced feature.
* You're already in a modal "are you sure" context.
* You're editing a synced pattern in a site editor context.
* You chose a rename option from an ellipsis menu.

Outside the ultimate safeguard, being able to see whether a block has overrides or not, we should still be mindful of adding clicks. Ultimately this is not a EULA you're agreeing to. You may just be building out the pattern initially, providing a name, creating a typo and wanting to fix that, knowing full well that no other context in your site is yet even using the pattern.

At the very least, I'd _start_ without the checkbox. It's always easier to add after the fact, than it is to remove.

File: packages/patterns/src/components/allow-overrides-modal.js
Comment (talldan) [inline packages/patterns/src/components/allow-overrides-modal.js:37]: There seems to be a focus loss here because of the way the button changes to a toggle 🤔

File: packages/patterns/src/components/pattern-overrides-controls.js
Comment (talldan) [inline packages/patterns/src/components/pattern-overrides-controls.js]: Feels like this could be `setShowAllowOverridesModal( true )` instead of `updateBindings( true )`.

File: packages/patterns/src/components/pattern-overrides-controls.js
Comment (talldan) [inline packages/patterns/src/components/pattern-overrides-controls.js]: To me it feels like this isn't needed. Is there ever a case where you toggle the `ToggleControl` and have the modal shown?

As noted in the other comment, perhaps the button could call `setShowAllowOverridesModal( true )` directly.

**Comment by fabiankaegy** [comment]
@jasmussen Yeah, I won't block merging this without the checkbox. However here are my reasons why I disagree: 

> You're editing a synced pattern in a site editor context.
This is not necessarily correct. From any post/page you can click on the "Edit Original" button in the toolbar and get to the editing state. So this is something that can be done by anyone. 

Even non-administrator roles that cannot access the site editor at all can edit synced patterns this way. So I do think this is a much more common action. 

Here you can see the flow for an editor user role:

https://github.com/WordPress/gutenberg/assets/20684594/53f0358c-f4f5-43d1-a565-5f6af65bcd10


> You chose a rename option from an ellipsis menu.

Given how renaming was a headlining feature for both 6.4 & 6.5 and the prevalence of users I see on a day-to-day basis that I would describe as non-technical that use the renaming feature I would consider the renaming action as a common occurrence. Not a super advanced feature. 

> You're already in a modal "are you sure" context.

If I recall the conversations about why the renaming option was moved into a modal correctly this was purely because of accessibility concerns with having it work inline like in Figma for example. I wouldn't necessarily describe the modal as a "are you sure" pattern given how many modals we have in the editor these days. 

> Outside the ultimate safeguard, being able to see whether a block has overrides or not, we should still be mindful of adding clicks. Ultimately this is not a EULA you're agreeing to. You may just be building out the pattern initially, providing a name, creating a typo and wanting to fix that, knowing full well that no other context in your site is yet even using the pattern.

To me, the action of renaming is one that someone should really carefully consider because it is not possible for the user to actually gauge the impact of their change. If we could have a way to indicate to the user how often this pattern with overrides is already used that would make this easier. But given that we cannot know that I personally would want to optimize this much more for the case where there are already hundreds of instances out there, vs the case where someone is creating a pattern from scratch and had a typo in the name. 

And when I think about that use case where someone already has hundreds of overrides I do think that the additional click to make sure someone is really aware of the consequences is worth including :)

**Review by talldan — COMMENTED** [review]
Looks good! Feels very close. I left a few review comments.

I also found one edge case:
1. Create a pattern with a paragraph
2. Enable overrides for the paragraph, giving it a name
3. Rename the paragraph, removing the name entirely
4. Save the pattern and insert it somewhere
5. Observe that the paragraph is editable in the instance (because it still has bindings), but when you view the post the bindings don't work (because there's no name)

I think either the binding attribute needs to be removed when removing a name (which might be preferable), or some of the code elsewhere is updated to consider a block overridable only when it has the bindings attribute and a name.

File: packages/block-editor/src/components/block-rename/modal.js
Comment (talldan) [reply packages/block-editor/src/components/block-rename/modal.js]: cc @fabiankaegy, as I see you also mentioned it in your feedback above. Let me know if you have any thoughts on the wording.

File: packages/patterns/src/components/allow-overrides-modal.js
Comment (talldan) [reply packages/patterns/src/components/allow-overrides-modal.js]: I guess "Ingredients" is also not ideal, seeing as users can't override the list block.

Maybe "Recipe Title", "Recipe Description" would be better choices.

File: packages/block-editor/src/components/block-rename/modal.js
Comment (fabiankaegy) [reply packages/block-editor/src/components/block-rename/modal.js]: I like your change here though I don't think it solves the concern I have. To me the help text here is just not noticeable / disruptive enough for how string the consequences may be :)

File: packages/patterns/src/components/pattern-overrides-controls.js
Comment (kevin940726) [reply packages/patterns/src/components/pattern-overrides-controls.js]: Oops, yeah. I think it's from an older exploration.

File: packages/patterns/src/components/allow-overrides-modal.js
Comment (kevin940726) [reply packages/patterns/src/components/allow-overrides-modal.js:37]: Should be fixed now.

**Comment by SaxonF** [comment]
> What's the reason we are showing the block connection icon when an overwritable block is selected in the contentOnly locked mode? Seems odd to have this block icon shown here especially since it isn't very descriptive.

+1 if we can avoid this I would. This is not a perfect solution for block connections either so will likely be iterated on.

**Comment by richtabor** [comment]
> What's the reason we are showing the block connection icon when an overwritable block is selected in the contentOnly locked mode? Seems odd to have this block icon shown here especially since it isn't very descriptive.
> +1 if we can avoid this I would. This is not a perfect solution for block connections either so will likely be iterated on.

It's a bit of an aside, but perhaps this could work. Denoting that this is bound to something else, much like template parts/synced patterns. 

CleanShot 2024-04-10 at 11 44 42

**Comment by fabiankaegy** [comment]
@richtabor i wonder whether it should have the purple color at all. Because the purple is always meant to represent global elements. And the override is exactly the opposite 🤔😅

**Comment by richtabor** [comment]
We can certainly iterate from here, but I did note that it's unexpected to have two different interactions for enabling and disabling overrides, and especially two different interactions for enabling. 

Can we simplify to this below (don't mind the actual design metrics, but the flow): 

https://github.com/WordPress/gutenberg/assets/1813435/5feb057f-7872-42a6-99b4-5a9610baab41

Removing the connection between `metadata.name` and the UI changing from a button to a toggle (removing the toggle altogether). Even if my block has a `metadata.name` value, I should still see the same "Allow overrides" button. 

I click "Allow overrides" and a modal renders with the name field. I am required to add a name if there is not one. If one already exists, I see it in the input field already. 

I select "Allow overrides" to confirm my override. When I press "Allow overrides" on the modal, the binding is applied—just like when I first created a binding initially. 

Now I see the secondary button in the Inspector, but it renders "Disable overrides".

When I select that button, I see a "Disable overrides" modal to confirm my decision. The help text should state what will happen to my existing overrides. There is no name field on this modal. When I press "Disable overrides" on the modal, the binding is removed.

**Comment by richtabor** [comment]
> @richtabor i wonder whether it should have the purple color at all. Because the purple is always meant to represent global elements. And the override is exactly the opposite 🤔😅

Yea, perhaps that doesn't work 1:1. Thoughts on using purple for bindings/overrides @jameskoster?

I suppose this is where bindings and overrides gets muddied a bit. As blocks with bounded attributes have values derived from "elsewhere"—whether or not that's "global" is another question.

**Comment by jameskoster** [comment]
Agree it's not perfect, but on balance I don't mind it too much because parts of the block _are_ still synced (styles). Worth noting perhaps that overrideable layers in Figma's component instances are always purple. 

That said, maybe the icon remains purple until there's an override, and then switches to the regular color. Could be worth a try? I worry it will be tricky to keep track of all this color changing though, across list view, inspector panel(s), etc.

**Comment by SaxonF** [comment]
@richtabor if we feel the button communicates both state and action (enabled and disable) then I'm fine with that approach.

**Comment by SaxonF** [comment]
Using colour only to indicate whether a block has been overridden will be an accessibility issue. If we can establish a common pattern we can use for styles (locally set vs global) and blocks that would be preferable. Eg a dot to indicate a change

**Comment by richtabor** [comment]
> @richtabor if we feel the button communicates both state and action (enabled and disable) then I'm fine with that approach.

I think it's fine yea.

**Comment by kevin940726** [comment]
> I think either the binding attribute needs to be removed when removing a name (which might be preferable), or some of the code elsewhere is updated to consider a block overridable only when it has the bindings attribute and a name.

This is fixed in `isOverridableBlock` to also check for `metadata.name`. This better matches what we want in https://github.com/WordPress/gutenberg/issues/58601.

**Review by talldan — APPROVED** [review]
This is working well for me. I'd be keen to get this merged to unblock other work and then follow-up on any of the smaller parts that may need to be changed.

**Pull request merged**
